publish.sh

```diff
@@ -2,73 +2,125 @@
 
 set -x
 
-if [[ "${CIRCLE_TAG}" == *dev ]]; then
+artifact_dir=~/artifacts/linux
+for arch in x86_64 arm64; do
+    if [ ! -f "${artifact_dir}/extension-${arch}.zip" ]; then
+        echo "${arch} extension does not exist, cannot publish."
+        exit 1;
+    fi
+done
+
+VERSION="${CIRCLE_TAG:-$(make version)}"
+# lambda layer names must match a regex: [a-zA-Z0-9-_]+)|[a-zA-Z0-9-_]+
+# turn periods into dashes and cry into your coffee
+VERSION=$(echo ${VERSION} | tr '.' '-')
+
+if [[ "${VERSION}" == *dev ]]; then
     EXTENSION_NAME="honeycomb-lambda-extension-dev"
 else
     EXTENSION_NAME="honeycomb-lambda-extension"
 fi
 
-REGIONS_NO_ARCH=(eu-north-1 us-west-1 eu-west-3 ap-northeast-2 sa-east-1 ca-central-1 af-south-1
-                ap-east-1 eu-south-1 me-south-1 ap-southeast-3 ap-northeast-3)
-REGIONS_WITH_ARCH=(ap-south-1 eu-west-2 us-east-1 eu-west-1 ap-northeast-1 ap-southeast-1
-                   ap-southeast-2 eu-central-1 us-east-2 us-west-2)
+# Region list update from AWS Lambda pricing page as of 2022/10/12
+#
+# regions with x86_64 only support
+# REGIONS_NO_ARCH=(me-central-1) # listed on pricing page, but we need grant our publishing credentials permission to publish there first
```

Comment:
We'll need to do the role permissions grant to publish in Middle East (UAE) before adding this region to the x86_64 only list.